test/framework/logs.go

```diff
@@ -99,19 +98,23 @@ func DumpPodLogs(logger Logger, podLogs PodSetLogs) {
 	}
 }
 
-// MustFollowPodLog attaches to the pod log stream, reads it until the pod is
-// dead or an error happens while reading.
+// FollowPodLog attaches to the pod log stream, reads it until the pod is dead
+// or an error happens while reading.
 //
-// If an error happens when fetching pod's Stream() this function calls t.Fatal
-// while if a failure happens during pods log read the error is sent back to
-// the caller through an error channel.
-func MustFollowPodLog(t *testing.T, pod corev1.Pod) (<-chan string, <-chan error) {
-	client := MustNewClientset(t, nil)
+// If an error happens when fetching pod's Stream() this function returns
+// immediately. If a failure happens during pods log read the error is sent back
+// to the caller through an error channel.
+func FollowPodLog(pod corev1.Pod) (<-chan string, <-chan error, error) {
+	client, err := NewClientset(nil)
```

Comment:
It's better to reuse the client (pass it as a parameter) and not to recreate it from scratch in every function.

Reply:
done.